MapboxCoreNavigation/RouteStep.swift

```diff
@@ -38,4 +41,30 @@ extension RouteStep {
     open var lastInstruction: SpokenInstruction? {
         return instructionsSpokenAlongStep?.last
     }
+    
+    /**
+     Returns true if the current route step contains a tunnel.
+     */
+    var containsTunnel: Bool {
+        guard let intersections = intersections else { return false }
+        for intersection in intersections {
+            if intersection.outletRoadClasses?.contains(.tunnel) == true {
+                return true
+            }
+        }
+        return false
+    }
+
+    /**
+     Getter/Setter to retrieve and store all intersection distances.
+     */
+    var intersectionDistances: [CLLocationDistance] {
```

Comment:
I'm unsure if this is the best place to store the property. What about putting this on RouteProgress?

Reply:
@bsudekum we only need to access the intersection distances for the `current step`, that's why I put that property here. In `RouteProgress` is looks out of place -- for instance, we need to call the variable `currentStepDistances`. `RouteProgress` has a variable, `currentStep` which is of type, `RouteStep` so it just made sense to put the current step's intersection distances in there. Or what do you think?

Reply:
While it would be desirable to put this property on RouteStep itself, the problem is the RouteStep is immutable. You’ve worked around this issue by accessing a global variable, but now it looks like it’s possible to get the `intersectionDistances` of a step other than the current step, which will unexpectedly return the intersection distances of the current step.

Due to the immutability of Route, RouteLeg, and RouteStep, we’ve built up a parallel structure of *Progress classes that can store transient state like this.